### ISSUE TITLE
 maas_local check ansible_fqdn is an issue

### ISSUE THREAD

**Issue by BjoernT** [rcbops/rpc-extras#9]
Some checks fail in maas_local due to playbook changes :

nova_compute check for instance:

9.0.1
   check_details: file=nova_service_check.py,args=--host,args={{ ansible_hostname }},args={{ internal_vip_address }}
9.0.5
  check_details: file=nova_service_check.py,args=--host,args={{ ansible_fqdn }},args={{ internal_vip_address }}

This change caused all checks to go fatal because the nova service name is installed using the short hostname, the actual maas check is looking for a FQDN name. So we either revert this change or change the maas checks to look for short hostnames first and then the FQDN hostname.

**Comment by b3rnard0** [comment]
We are also tracking this here: https://trello.com/c/SwJqrVzY/460-rpc-extras-maas-local-check-ansible-fqdn-is-an-issue

**Comment by miguelgrinberg** [comment]
@BjoernT let me see if I understand. This isn't a problem introduced here in rpc-extras, same issue exists in the 9.0.5 version of the integrated maas plays?

**Comment by BjoernT** [comment]
Yes it does exist in older releases too.
But I can't file it as os-openstack-ansible since it was generalized, the Rackspace specifics got removed.

**Comment by miguelgrinberg** [comment]
Yes, that's fine, just wanted to make sure.

**Comment by mattt416** [comment]
It looks like the check for neutron_linuxbridge_agent_check is also affected by this.

**Comment by miguelgrinberg** [comment]
@mattt416 so it looks like you made the change from `ansible_hostname` to `ansible_fqdn`. Here is the commit: https://github.com/stackforge/os-ansible-deployment/commit/d7216eccb6e7733f1079e09d99b04ca58634da19.

Unfortunately I can't find issue #452, that probably refers to the old ansible-lxc-rpc repo, and the issues there were removed for some reason.

I can't find any information regarding the problem that made you change this, I really don't understand why `ansible_hostname` was not giving you the correct thing in the first place.

I can revert these back to `ansible_hostname`, but what am I breaking if I do that?

**Comment by mattt416** [comment]
@miguelgrinberg Correct -- I appear to have broken this.  :)  The original issue was that we were using `ansible_hostname` everywhere, which was problematic if someone had deployed nodes w/ a proper FQDN.  The change made was fine except for `nova_compute_check` and `neutron_linuxbridge_agent_check` which use the FQDN:

```
root@aio1_utility_container-0cdd7735:~# nova service-list
+----+------------------+--------------------------------------------+----------+---------+-------+----------------------------+-----------------+
| Id | Binary           | Host                                       | Zone     | Status  | State | Updated_at                 | Disabled Reason |
+----+------------------+--------------------------------------------+----------+---------+-------+----------------------------+-----------------+
| 1  | nova-cert        | aio1_nova_cert_container-00a2f79d          | internal | enabled | up    | 2015-03-04T10:25:11.000000 | -               |
| 7  | nova-conductor   | aio1_nova_conductor_container-4a2b2c2d     | internal | enabled | up    | 2015-03-04T10:25:09.000000 | -               |
| 10 | nova-scheduler   | aio1_nova_scheduler_container-c5f35b14     | internal | enabled | up    | 2015-03-04T10:25:07.000000 | -               |
| 13 | nova-consoleauth | aio1_nova_spice_console_container-04b4da91 | internal | enabled | up    | 2015-03-04T10:25:12.000000 | -               |
| 16 | nova-compute     | aio1.rpc.test.com                        | nova     | enabled | up    | 2015-03-04T10:25:05.000000 | -               |
+----+------------------+--------------------------------------------+----------+---------+-------+----------------------------+-----------------+
root@aio1_utility_container-0cdd7735:~#
```

```
root@aio1_utility_container-0cdd7735:~# neutron agent-list
+--------------------------------------+--------------------+----------------------------------------+-------+----------------+---------------------------+
| id                                   | agent_type         | host                                   | alive | admin_state_up | binary                    |
+--------------------------------------+--------------------+----------------------------------------+-------+----------------+---------------------------+
| 21cf74a9-62ae-46f9-9135-815c55559eba | Linux bridge agent | aio1.rpc.test.com                    | :-)   | True           | neutron-linuxbridge-agent |
| 2579ffdf-8446-4e56-9ad4-f21973690b0a | Metering agent     | aio1_neutron_agents_container-ea8c3a07 | :-)   | True           | neutron-metering-agent    |
| 29392754-27a9-4f95-84da-695f84b345b4 | DHCP agent         | aio1_neutron_agents_container-ea8c3a07 | :-)   | True           | neutron-dhcp-agent        |
| 49fc7608-dedf-4a53-ab42-ff78badd5fca | L3 agent           | aio1_neutron_agents_container-ea8c3a07 | :-)   | True           | neutron-l3-agent          |
| e9cd0c90-8759-40ef-926c-f054c8cc5ef0 | Metadata agent     | aio1_neutron_agents_container-ea8c3a07 | :-)   | True           | neutron-metadata-agent    |
| f5f45e42-3b47-4f49-a217-f8108d83c8e6 | Linux bridge agent | aio1_neutron_agents_container-ea8c3a07 | :-)   | True           | neutron-linuxbridge-agent |
+--------------------------------------+--------------------+----------------------------------------+-------+----------------+---------------------------+
root@aio1_utility_container-0cdd7735:~#
```

**Comment by miguelgrinberg** [comment]
@mattt416 has kindly volunteered to remove the offending `ansible_fqdn` calls and test.

**Comment by mattt416** [comment]
I think I better understand this issue now.  As @BjoernT said, we were originally using {{ ansible_hostname }} (short name).  We then received an issue from QE stating that this should be {{ {{ ansible_fqdn }} (FQDN).  At the time, I booted an AIO cloud server w/ an FQDN hostname, verified what they claimed, and made the change.

Having looked at this again, I noticed that a freshly built cloud server AIO w/ FQDN hostname had FQDNs listed for those non-containerised hosts listed in a `nova service-list` and `neutron agent-list`.  Chatting w/ @andymcc, he was seeing the reverse (short names rather than FQDNs).  Upon further investigation, the hostname didn't appear to be set correctly on my AIO (the FQDN was listed in /etc/hostname and a `hostname -s` was returning the FQDN), where-as on @andymcc's installation everything looked correct.  I will assume that the QE team's hosts were originally configured incorrectly which is why they requested that we adjust these plays.

I am currently looking at ways to handle both situations so that we don't have to go back to deployers and tell them to fix their hostnames (as ideal as that would be).

**Comment by b3rnard0** [comment]
Need to determine if this will be an icehouse backport.

@claco @mancdaz @cloudnull

**Comment by mattt416** [comment]
Backport to juno handled in https://review.openstack.org/#/c/169668/.

**Comment by b3rnard0** [comment]
This ended up being released in 10.1.3 here:

https://bugs.launchpad.net/openstack-ansible/+bug/1439083
